compiler/rustc_expand/src/mbe/macro_rules.rs

```diff
@@ -183,6 +183,46 @@ impl<'matcher> Tracker<'matcher> for NoopTracker {
     }
 }
 
+#[instrument(skip(cx, tts))]
+pub fn expand_token_stream<'cx>(
+    cx: &'cx mut ExtCtxt<'_>,
+    sp: Span,
+    arm_span: Span,
+    node_id: NodeId,
+    name: Ident,
+    tts: TokenStream,
+) -> Box<dyn MacResult + 'cx> {
```

Comment:
I'd love to know if there is already a standard way to expand a token stream. I couldn't find one, so I did some refactoring here.

Reply:
You can just return a token stream from a built-in macro, end rustc expand will parse it appropriately, see the `quote` example in `compiler\rustc_builtin_macros\src\lib.rs`.
But then you lose the opportunity to return `DummyResult::any` for diagnostics.
So what you are doing here is fine.

Perhaps we need some better interface for retuning *either* a token stream, or an equivalent of `DummyResult::any` in general.